src/include/migraphx/builtin.hpp

```diff
@@ -63,6 +63,13 @@ struct param
     }
 };
 
+struct add_return
+{
+    std::string name() const { return "return"; }
```

Comment:
The name should be `@return` as all the builtins start with `@`.

Reply:
instruction name cannot start with "@", see 

instruction_ref program::add_instruction(const operation& op, std::vector<instruction_ref> args)
{
    return insert_instruction(impl->instructions.end(), op, std::move(args));
}
instruction_ref program::insert_instruction(instruction_ref ins,
                                            const operation& op,
                                            std::vector<instruction_ref> args)
{
    assert(std::all_of(
               args.begin(), args.end(), [&](instruction_ref x) { return has_instruction(x); }) &&
           "Argument is not an exisiting instruction");
    assert(not starts_with(op.name(), "@"));

Reply:
You can either change the symbol or change the assertion to make an exception for return.

Reply:
renamed the name to @return and add throw exception to compute function.